base/combinatorics.jl

```diff
@@ -2,38 +2,40 @@
 
 # Factorials
 
-const _fact_table64 = Vector{Int64}(undef, 20)
-_fact_table64[1] = 1
-for n in 2:20
-    _fact_table64[n] = _fact_table64[n-1] * n
-end
+const _shifted_fact_table64 = [1; cumprod(1:Int64(20))]  # tables shifted by 1 so factorial(0) can be looked up; maybe OffsetArray better?
 
-const _fact_table128 = Vector{UInt128}(undef, 34)
-_fact_table128[1] = 1
-for n in 2:34
-    _fact_table128[n] = _fact_table128[n-1] * n
-end
+const _shifted_fact_table128 = [1; cumprod(1:UInt128(34))]
 
-function factorial_lookup(n::Integer, table, lim)
+const _fact_table64 = [cumprod(1:Int64(20))]  # kept just for gamma function (Base._fact_table64) in SpecialFunctions.jl, notused here
+
+@noinline function factorial_lookup_helper(n::Integer, table, lim)  # non-fast-path, almost identical to the old code
     n < 0 && throw(DomainError(n, "`n` must not be negative."))
     n > lim && throw(OverflowError(string(n, " is too large to look up in the table")))
-    n == 0 && return one(n)
-    @inbounds f = table[n]
+    # no longer needed because of shift in table:   n == 0 && return one(n)
```

Comment:
This comment is not useful (because someone reading the new code has no idea about the old code).